### ISSUE TITLE
Array of dates are not encoded correctly as a query parameter

### ISSUE THREAD

**Issue by TimAEllis** [vapor/vapor#2909]
### Describe the bug

Attempting to encode an array of dates as a query param ignores the `dateEncodingStrategy` set.

### To Reproduce

Very basic example extracted from my codebase

```swift

func test(client: Client) async throws {
    // Breaking example
    let dates: [Date] = [
        Date(),
    ]
    try await client.get("example.com/multi-dates") { req in
        let encoder = URLEncodedFormEncoder(
            configuration: URLEncodedFormEncoder.Configuration(
                arrayEncoding: .values, 
                dateEncodingStrategy: .iso8601
            )
        )
        try req.query.encode([
            "dates": dates
        ], using: encoder)
        // This shows that the date is actually still seconds formatted instead of iso8601
        print("query: \(req.query)")
    }

    // Passing example
    let singleDate = Date()
    try await client.get("example.com/single-date") { req in
        let encoder = URLEncodedFormEncoder(
            configuration: URLEncodedFormEncoder.Configuration(
                arrayEncoding: .values, 
                dateEncodingStrategy: .iso8601
            )
        )
        try req.query.encode([
            "dates": singleDate
        ], using: encoder)
        // Corrected formatted date this time
        print("query: \(req.query)")
    }
}
```

1. Add Vapor package
2. Send run above block
3. Look at console when it is printed

### Expected behavior

Breaking example should output something like:
```
example.com/multi-dates?dates=2022-11-11&dates=2022-11-12 //...
```

But it actually outputs something like:
```
example.com/multi-dates?dates=12345&dates=23456 //...
```

### Environment

* Vapor Framework version: 4.67.1
* Vapor Toolbox version: 18.5.1 (but integrated into larger project via SPM)
* OS version: macOS Ventura 13.0 (22A380)

### Additional context

~ nil

**Comment by Joannis** [comment]
I know this kind of bug, Date is a really odd implementation in Swift. Commonly used, but also commonly needing an exception in Decoder implementations. It's likely that `URLEncodedFormEncoder` forgot to do the type checks for arrays.

**Comment by Joannis** [comment]
Didn't take very long! Thanks for the bugreport @TimAEllis

**Comment by TimAEllis** [comment]
With #2914 merged and available in [4.67.4](https://github.com/vapor/vapor/releases/tag/4.67.4), I think we can safely close this issue off as completed
